src/shell-app.c

```diff
@@ -1428,6 +1420,24 @@ shell_app_create_custom_launcher_with_name (ShellApp *app,
 
   new_path = g_build_filename (g_get_user_data_dir (), "applications", shell_app_get_id (app), NULL);
 
+  if (!g_file_test (new_path, G_FILE_TEST_EXISTS))
+    {
+      /* Create a new 'X-Endless-CreatedBy' key to indicate the file was created
+       * by eos-desktop */
+      g_key_file_set_string (keyfile,
```

Comment:
That defeats the purpose of the patch because I noticed this problem initially when using *xdg-app* formerly on the user folder. It installed desktop files into *~/.local/share/applications* and upon removing the files from the desktop grid, they would be deleted from the folder too, losing the desktop file.
So that's why we need this fix mainly. An alternative may be to use a different (not so standard) folder to hold the renamed desktop files, in which case we can even drop the special key.